xla/service/gpu/gpu_performance_model.cc

```diff
@@ -70,13 +71,138 @@ absl::Duration ReadTime(const GpuDeviceInfo& gpu_device_info,
   return absl::Seconds(n_bytes_total / bw);
 }
 
+int64_t GetMaxNumberOfChannels(
+    GpuPerformanceWithCollectiveModel::CollectiveAlgo algorithm) {
+  int64_t max_nchannels = 0;
+  switch (algorithm) {
+    // Tree and Ring algos share the same max channel number.
+    case GpuPerformanceWithCollectiveModel::RING:
+    case GpuPerformanceWithCollectiveModel::TREE:
+      max_nchannels = GpuPerformanceWithCollectiveModel::max_nchannels_ring;
+      break;
+  }
+  const char* env = std::getenv("NCCL_MAX_NCHANNELS");
```

Comment:
I didn't see any references to this, but looks like its a new env var in NCCL 2.18. Just a comment, no AI.